### PR TITLE
Improve test suite

### PR DESCRIPTION
**Pull request by pascalbaljet**
- Added GitHub Actions workflow
- Bumped `saloonphp/laravel-http-sender` dependency to support Laravel 11
- Bumped Laravel 10 requirement to `^10.32` as it requires `getGlobalMiddleware()`
- Removed unneeded `guzzlehttp/guzzle` dependency 
- Streamlined `EventServiceProvider` into `ServiceProvider`
- Added tests for `EventHelper`, `PendingRequestMixin` and `ServiceProvider`
- The added Guzzle middleware now has a name for easier debugging and testing

### CODE REVIEW

File: tests/PendingRequestMixinTest.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    /** @var PendingRequest $pendingRequest */
+    $pendingRequest = Http::withHeaders(['X-Test' => 'true'])->log();
+
+    /** @var HandlerStack $handler */
+    $handler = $pendingRequest->getOptions()['handler'];
+    expect($handler)->toBeInstanceOf(HandlerStack::class);
+
+    // It adds a logger for each format (see config('http-client-global-logger.format'))
+    expect($handler->hasHandler('http-client-global-logger-0'))->toBeTrue()
```
Comment (onlime) [inline tests/PendingRequestMixinTest.php]: @pascalbaljet `hasHandler()` does not expect middleware name. Shouldn't it be something like:

```php
    expect($handler)->toBeInstanceOf(HandlerStack::class)
        ->and($handler->hasHandler())->toBeTrue()
        // It adds a logger for each format (see config('http-client-global-logger.format'))
        ->and($handler->findByName('http-client-global-logger-request'))->toBeInt()
        ->and($handler->findByName('http-client-global-logger-response'))->toBeInt();
```

but `HandlerStack::findByName()` is `private`, so currently don't know how to check that. Also, the keys are `request|response` in default config.

**Review by onlime — CHANGES_REQUESTED** [review]
great PR! 👏 Thanks @pascalbaljet 
just needs one small refinement...

File: tests/PendingRequestMixinTest.php (hunk 1 above)
Comment (pascalbaljet) [reply tests/PendingRequestMixinTest.php]: Fixed in d5f7899. Also investigating another solution now.

Comment (pascalbaljet) [reply tests/PendingRequestMixinTest.php]: Another option would be to use something like `spatie/invade` and check the private `$stack` property:

```php
$stack = invade($handler)->stack;
$middlewareNames = Arr::pluck($stack, 1);

expect($middlewareNames)->toContain('http-client-global-logger-0')
    ->and($middlewareNames)->toContain('http-client-global-logger-1')
    ->and($middlewareNames)->not->toContain('http-client-global-logger-2');
```

Comment (onlime) [reply tests/PendingRequestMixinTest.php]: yes, we all love `spatie/invade`! 😄 prefer this solution

Comment (pascalbaljet) [reply tests/PendingRequestMixinTest.php]: Fixed in 51d1c9c

Comment (onlime) [reply tests/PendingRequestMixinTest.php]: I have simplified it further in 7fbada6 by invading `HandlerStack::findByName()`

**Pull request merged**